File: SampleProjects/codepilot/src/Class/Tables/CircleTable.ts
import { Food } from "../Food";
import { FoodOnTable } from "../FoodOnTable";
import { ITable } from "../Interfaces/ITable";

export class CircleTable implements ITable{
    x: number;
    y: number;
    isHovered: Boolean;
    tableId: number;
    isSelected: Boolean;
    color: String;
    tableStatus: String;

    foods: FoodOnTable[];

    radius: number;

    constructor(tableId: number,x: number, y: number, radius: number, tableStatus: String){
        this.x = x;
        this.y = y;
        this.radius = radius;
        this.foods = [];
        this.tableStatus = tableStatus;
        this.tableId = tableId;
    }

    setIsHovered(value: boolean): void {
        this.isHovered = value;
        return;
    }

    isMouseInRange(x: number, y: number): boolean {
        const distance = Math.sqrt((x - this.x) ** 2 + (y - this.y) ** 2);
        return distance < this.radius;
    }
    

    draw(context): void {
        context.beginPath();
        context.arc(this.x, this.y,this.radius, 0, Math.PI * 2,false);

        if(this.tableStatus === "unlocked"){
            context.fillStyle = "#D9D9D9";
        }else if(this.tableStatus === "locked"){
            context.fillStyle = "#66DD56";
        }else if(this.tableStatus === "occupied"){
            context.fillStyle = "rgba(209, 150, 133, 0.69)";
        }else{
            throw Error("Not recognized behavior!");
        }

        context.fill();
        if(this.isSelected){
            context.beginPath();
            context.arc(this.x, this.y,this.radius, 0, Math.PI * 2,false);
            context.strokeStyle = "rgba(123, 166, 180, 1)";
            context.lineWidth = 5;
            context.stroke();
        }
        else if(this.isHovered){
            context.beginPath();
            context.arc(this.x, this.y,this.radius, 0, Math.PI * 2,false);
            context.strokeStyle = "rgba(173, 216, 230, 1)";
            context.lineWidth = 5;
            context.stroke();
        }
    }
}

File: SampleProjects/BlazorFE/BlazorApp1/Class/CircleTable.cs
namespace BlazorApp1.Class
{
    using System;
    using System.Collections.Generic;

    public class CircleTable : ITable
    {
        public int X { get; set; }
        public int Y { get; set; }
        public bool IsHovered { get; set; }
        public int TableId { get; set; }
        public bool IsSelected { get; set; }
        public string? Color { get; set; }
        public string? TableStatus { get; set; } // "locked", "unlocked", "occupied"
        public string? ShapeId { get; set; }
        public List<FoodOnTable> Foods { get; set; } = new List<FoodOnTable>();
        public string? ImgURL { get; set; }
        public string? ShapeName { get; set; }
        public double Radius { get; set; }

        // Constructor
        public CircleTable(int tableId, string shapeId, int x, int y, double radius, string tableStatus, string ImgURL, string ShapeName)
        {
            X = x;
            Y = y;
            Radius = radius;
            Foods = new List<FoodOnTable>();
            TableStatus = tableStatus;
            TableId = tableId;
            ShapeId = shapeId;
			this.ImgURL = ImgURL;
			this.ShapeName = ShapeName;
		}

        public void SetIsHovered(bool value)
        {
            IsHovered = value;
        }

        public bool IsMouseInRange(double x, double y)
        {
            var distance = Math.Sqrt((x - X) * (x - X) + (y - Y) * (y - Y));
            return distance < Radius;
        }

        public void Draw(object context)
        {
            // Implement drawing logic using Blazor's Canvas or other graphics library
            // Placeholder for implementation
            throw new NotImplementedException();
        }
    }

}
